mpi-proxy-split/mpi-wrappers/generate-mpi-fortran-wrappers.py

```diff
@@ -25,7 +25,7 @@
 declarations_file.close()
 
 def print_mpi_type_struct():
-  print("#ifdef CRAY_MPICH_VERSION")
+  print("#if MPICH_NUMVERSION >= MPICH_CALC_VERSION(3,4,0,0,2) && defined(CRAY_MPICH_VERSION)")
```

Comment:
Why MPICH 3.4?  Can we add a short comment about what changed in that version.

Reply:
@JainTwinkle Any idea why checking MPICH_CALC_VERSION(3,4,0,0,2)? I saw this checking was first added for api_type_struct for build mana on Perlmutter.

Reply:
MPICH 3.0 version has already used the same API. Why we choose version 3.4a2?

Reply:
Hi @dahongli and @gc00,
I was getting a compile-time error on Perlmutter with the existing macro that checks for `CRAY_MPICH_VERSION`. This is required to differentiate between a cray and a non-cray platform. However, that was not sufficient for Perlmutter as it uses the else part signature (`(MPI_Datatype*) array_of_types` argument without the `const`). So, at that time, I needed to differentiate between Cori's cray-mpich and Perlmutter's cray-mpich, to ensure that the else section API is being used when running on Perlmutter.
Feel free to improve the if...else condition.